syntax = "proto3";

service Booking {
  // Get all bookings from database
  rpc GetListBookings (EmptyMessage) returns (stream Book);

  //Get All bookings from specified user
  rpc GetListBookingsFromUser (UserId) returns (stream Book);

  //Add booking in the database
  //Returns string message
  rpc AddBooking (Reservation) returns (AddBookingReturnMessage);
}

message EmptyMessage{}

message Book {
  message Dates{
    string date = 1;
    repeated string movies = 2;
  }

  string userid = 1;
  repeated Dates dates = 2;
}

message UserId{
  string userid = 1;
}

message Reservation{
  string userid = 1;
  string movie = 2;
  string date = 3;
}

message AddBookingReturnMessage{
  int32 code = 1;
  string message = 2;
}